apps/stripe/src/app/api/stripe/webhook/stripe-object-handlers/stripe-payment-intent-handler.test.ts

```diff
@@ -0,0 +1,255 @@
+
+import { StripePaymentIntentHandler } from "./stripe-payment-intent-handler";
+
+describe("StripePaymentIntentHandler", () => {
+  describe.todo("checkIfEventIsSupported", () => {
+    it.each([
+      {
+        event: getMockedPaymentIntentSucceededEvent(),
+      },
+      {
+        event: getMockedPaymentIntentProcessingEvent(),
+      },
+      {
+        event: getMockedPaymentIntentRequiresActionEvent(),
+      },
+      {
+        event: getMockedPaymentIntentAmountCapturableUpdatedEvent(),
+      },
+      {
+        event: getMockedPaymentIntentPaymentFailedEvent(),
+      },
+    ])("should return true for supported event: $event.type", ({ event }) => {
+      const result = new StripePaymentIntentHandler().checkIfEventIsSupported(event);
+
+      expect(result).toBe(true);
+    });
+
+    it("should return false for unsupported event", () => {
+      const event = {
+        type: "payment_intent.created",
+      } as unknown as Stripe.Event;
+
+      const result = new StripePaymentIntentHandler().checkIfEventIsSupported(event);
+
+      expect(result).toBe(false);
+    });
+  });
+
+  describe("processPaymentIntentEvent", () => {
+    describe("type: payment_intent.succeeded", () => {
+      it.each([
+        createResolvedTransactionFlow("AUTHORIZATION"),
+        createResolvedTransactionFlow("CHARGE"),
+      ])(
+        "should resolve fields from Stripe event properly for '%s' flow",
+        (resolvedTransactionFlow) => {
+          const event = getMockedPaymentIntentSucceededEvent();
+
+          event.data.object.amount_received = 123_30;
+
+          const recordedTransaction = getMockedRecordedTransaction({
+            resolvedTransactionFlow,
+          });
+
+          const result = new StripePaymentIntentHandler().processPaymentIntentEvent({
+            event,
+            recordedTransaction,
+            stripePaymentIntentId: mockedStripePaymentIntentId,
+          });
+
+          const { type, amount, pspReference, time } = result
+            ._unsafeUnwrap()
+            .resolveEventReportVariables();
+
+          expect(type).toBe("CHARGE_SUCCESS");
+          // comes from mock
+          expect(amount.currency).toStrictEqual("USD");
+          // Converted to Saleor float
+          expect(amount.amount).toStrictEqual(123.3);
+          expect(pspReference).toStrictEqual(event.data.object.id);
+          expect(time).toStrictEqual("2025-02-01T00:00:00.000Z");
+        },
+      );
+    });
+
+    describe("type: payment_intent.processing", () => {
+      it.each([
+        {
+          resolvedTransactionFlow: createResolvedTransactionFlow("AUTHORIZATION"),
+          expectedType: "AUTHORIZATION_REQUEST",
+        },
+        {
+          resolvedTransactionFlow: createResolvedTransactionFlow("CHARGE"),
+          expectedType: "CHARGE_REQUEST",
+        },
+      ])(
+        "should resolve fields from Stripe event properly for $resolvedTransactionFlow flow",
```

Comment:
lets put `expectedType` to test name so actual business case is visible in test report